Documentation/In-depth/ComposerJson/Index.rst

```diff
@@ -0,0 +1,285 @@
+.. include:: /Includes.txt
+
+
+.. _project_composer.json:
+
+=============
+composer.json
+=============
+
+In Composer, everything is a package, but there are differences between
+a "project" and a "library" and the best practices that are used within
+the file :file:`composer.json` for a project or library.
+
+This page explains the basics for your **project** (TYPO3 installation)
+:file:`composer.json`, while :file:`composer.json` for libraries (or more
+specifically for TYPO3 extensions) is handled in TYPO3 Explained
+:ref:`t3coreapi:composer-json`
+
+In a Composer based TYPO3 installation, we typically have a file structure
+like this:
+
+
+.. code-block:: none
+
+   project-root/
+   ├── composer.json
+   ├── composer.lock
+   └── public
+       ├── fileadmin
+       ├── typo3
+       ├── typo3conf
+       └── ...
+
+You may have used the `typo3/cms-base-distribution` package (as explained in
+:ref:`install-via-composer`) which created a :file:`composer.json`
+for you or you may want to create a :file:`composer.json` for your project from
+scratch.
+
+.. _project_composer.json-example:
+
+Example composer.json
+=====================
+
+.. code-block:: json
+
+   {
+      "name": "vendor/my-typo3-project",
+      "description": "TYPO3 project",
+      "license": ["GPL-2.0-or-later"],
+      "type": "project",
+      "authors": [
+         {
+            "name": "Author name",
+            "email": "nouser@example.com"
+         }
+      ],
+      "require": {
+         "php": "^7.4",
+         "helhum/typo3-console": "^6.4.0",
+         "typo3/cms-backend": "^10.4.15",
```

Comment:
Not trying to be too picky here but we should use `^10.4` instead. Otherwise we need to explain why we require at least this specific version.